handler/src/main/java/io/netty/handler/timeout/WriteTimeoutHandler.java

```diff
@@ -70,6 +70,8 @@
 
     private final long timeoutNanos;
 
+    private Set<ScheduledFuture<?>> timeouts;
```

Comment:
what we also could do is make the WriteTimeoutTask basically a doubly-linked list and so have a ref to `prev` and `next` task. This way adding and removing would be O(1) for our use-case and we not need any extra data structure. Not sure if this is better or not though